features/admin.groups.v1/components/wizard/group-basics.tsx

```diff
@@ -263,16 +262,23 @@ export const GroupBasics: FunctionComponent<GroupBasicProps> = (props: GroupBasi
                                 "roleBasicDetails.roleName.validations.empty", { type: "Group" }) }
                             validation={ async (value: string, validation: Validation) => {
                                 let isGroupNameValid: boolean = true;
+                                let userstoreRegex: string = "";
 
                                 await validateGroupNamePattern().then((regex: string) => {
+                                    userstoreRegex = regex;
                                     isGroupNameValid = SharedUserStoreUtils.validateInputAgainstRegEx(value, regex);
                                 });
 
                                 if (!isGroupNameValid) {
                                     validation.isValid = false;
-                                    validation.errorMessages.push(t("console:manage.features.businessGroups" +
-                                        ".fields.groupName.validations.invalid",
-                                    { type: "group" }));
+                                    if (userstoreRegex === groupConfig.defaultUserstoreRegex) {
+                                        // If the userstore regex is the default regex, show the default error message.
+                                        validation.errorMessages.push(t(groupConfig?.groupPrimaryUserstoreRegexHint));
```

Comment:
Is showing a regex in the end user facing UI intentional?

Reply:
Yes. The regex will be shown when a custom regex is used